hw/arm/hercules.c

```diff
@@ -672,3 +676,75 @@ static void hercules_register_types(void)
     type_register_static(&hercules_type_info);
 }
 type_init(hercules_register_types)
+
+static void hercules_xx57_init(MachineState *machine, bool is_tms570)
+{
+    Object *dev;
+    MemoryRegion *sdram = g_new(MemoryRegion, 1);
+    DriveInfo *eeprom = drive_get(IF_MTD, 0, 0);
+    const char *file;
+
+    dev = object_new(TYPE_HERCULES_SOC);
+    qdev_prop_set_drive(DEVICE(dev), "eeprom",
+                        eeprom ? blk_by_legacy_dinfo(eeprom) : NULL,
+                        &error_abort);
+    object_property_set_bool(dev, is_tms570, "is_tms570", &error_fatal);
```

Comment:
regarding instance_init vs realize, is this appropriate?

Reply:
I think so, but I am also sure we'd get a lecture from upstream if it isn't :-) I don't know why, but for some reason I think property names are supposed to use `-` instead of `_`, so we might want to call it `is-tms570`

Reply:
got it

Reply:
Figured out we can just use object_new_with_props instead